### ISSUE TITLE
MacOS 10.15 boxes cannot upgrade all packages with brew

### ISSUE THREAD

**Issue by Haroon-Khel** [adoptium/infrastructure#3050]
ref https://github.com/adoptium/infrastructure/pull/3045

MacOS 10.15 is out of support https://endoflife.date/macos. As a temp fix for the sake of preventing playbook crashes I have skipped the `upgrade_all` task on our macos 10.15 and under boxes but we need a better solution

@sxa Are we able to have these boxes upgraded?

Our 10.14 and 10.15 boxes make up 7/9 of our x64 mac machines, the other 2 being 12.01

**Comment by sxa** [comment]
@andrew-m-leonard Any feelings on this. Are there reproducibility/compatibilty issues that you're aware with if we were to update the build machines to a later macos?

**Comment by andrew-m-leonard** [comment]
@sxa I don't believe so for reproducibility, as that is dependent purely on the Xcode version used.
Compatibility, can't say for sure, but i'd hope the Xcode compatibility shoud at least support Apple supported levels?

**Comment by sxa** [comment]
Note that https://github.com/adoptium/temurin-build/blob/3c6f770e3dcb5bb79a8b201a7a03a75ae63e2467/build-farm/platform-specific-configurations/mac.sh#L47 explicitly targets 10.9 (Does that need a C flag as well as C++?) although only for JDK17+ on x64, so it may be ok on those versions, but not for earlier ones.

I think we'll need to test this before moving up. @Haroon-Khel can you run a build on a JDK17/x64 machine with a later OSX and see if it can still run (via a grinder or other test run) on the earlier versions? It may not matter since it's out of support but it would be good to have the data on whether there would be a potential impact to customers before moving. Same test on 8 should be done too but it seems more likely that it won't work there.

**Comment by andrew-m-leonard** [comment]
I suspect that will need changing to whatever the new Apple min supported version is? Wondering whether we could just remove it, now 10.9 is deprecated?
